src/Tokenizer/Token.php

```diff
@@ -104,11 +104,18 @@ public static function getCastTokenKinds(): array
      *
      * @return int[]
      */
-    public static function getClassyTokenKinds(): array
+    public static function getClassyTokenKinds(bool $enumIsClassy = false): array
```

Comment:
to be honest, it doesn't look like a good solution to me (same concern for `isClassy`)

i would suggest that we either make input more generic (eg `exluceList: array`) or we just always return enum as classy (looking at [examples](https://www.php.net/manual/en/language.enumerations.examples.php) they may have standard methods), and in those very few places we want to not affect them, check `->isClassy() && !->isKind(T_ENUM)`)

Reply:
it is mainly a BC layer thing as `Token` is not internal, so not sure I can just change the behavior of the method

Reply:
fixed